Snowflake.Data/Core/FileTransfer/StorageClient/SFGCSClient.cs

```diff
@@ -527,12 +436,37 @@ private void HandleDownloadResponse(HttpWebResponse response, SFFileMetadata fil
         }
 
         /// <summary>
-        /// Handle file header error.
+        /// Handle file header error for presigned urls.
+        /// </summary>
+        /// <param name="ex">The file header exception.</param>
+        /// <param name="fileMetadata">The GCS file metadata.</param>
+        /// <returns>File Metadata</returns>
+        private SFFileMetadata HandleFileHeaderErrForPresignedUrls(WebException ex, SFFileMetadata fileMetadata)
+        {
```

Comment:
I would log here the problem as well

Reply:
Down there (I cannot comment on the line) is HandleUploadFileError. Let's put any problem from there to the logs as well.

Reply:
Sure, added logging to those functions